gems/aws-sdk-polly/lib/aws-sdk-polly/presigner.rb

```diff
@@ -30,35 +33,42 @@ class Presigner
       #     * `Aws::AssumeRoleCredentials`
       #     * `Aws::ECSCredentials`
       #
-      # @option options [required, string] :region
-      #   The region name, e.g. 'us-west-2'
+      # @option options [String] :region This option is deprecated, please use
+      #   :client instead. The region name, e.g. 'us-west-2'.
+      #
       def initialize(options = {})
-        @credentials = options.fetch(:credentials)
-        @region = options.fetch(:region)
+        if !(options.keys - [:credentials, :region, :client]).empty?
+          raise ArgumentError,
+                ':options may only contain :region and :credentials keys. '\
```

Comment:
I find this wording very confusing.  You're saying it may only contain :region or :credentials and then telling the user to use something other than those?